### PR TITLE
[14.0][MIG] stock_reserve: Migration to 14.0

### PR DESCRIPTION
**Pull request by victoralmau**
Superseed https://github.com/OCA/stock-logistics-warehouse/pull/1346

Migration to 14.0

Please @pedrobaeza and @CarlosRoca13 can you review it?

@Tecnativa TT41080

### CODE REVIEW

File: stock_reserve/models/stock_reserve.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def get_location_from_ref(self, ref):
         """
         try:
             location = self.env.ref(ref, raise_if_not_found=True)
-            location.check_access_rule("read")
+            location.check_access_rights("read", raise_exception=False)
```
Comment (pedrobaeza) [inline stock_reserve/models/stock_reserve.py]: The `try` is for being triggered when there's an access exception, but you have removed the `raise_exception` part.

**Review by CarlosRoca13 — APPROVED** [review]
Code review :+1:

**Comment by victoralmau** [comment]
Changes done.

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**